tests/php/sync/test_class.jetpack-sync-posts.php

```diff
@@ -757,8 +757,12 @@ function test_remove_sharedaddy_from_filtered_content() {
 		// this only applies to rendered content, which is off by default
 		Settings::update_settings( array( 'render_filtered_content' => 1 ) );
 
-		require_once JETPACK__PLUGIN_DIR . 'modules/sharedaddy/sharing.php';
-		require_once JETPACK__PLUGIN_DIR . 'modules/sharedaddy/sharing-service.php';
+		if ( class_exists( 'Sharing_Service' ) ) {
+			Sharing_Service::init();
+		} else {
+			require_once JETPACK__PLUGIN_DIR . 'modules/sharedaddy/sharing.php';
+			require_once JETPACK__PLUGIN_DIR . 'modules/sharedaddy/sharing-service.php';
```

Comment:
Should we `init` after requiring the files here?

Reply:
I should've elaborated on that, thanks for raising that question.

When `sharing-service.php` is included, it adds filters for `the_content` and `the_excerpt`. The file used to call them directly, outside of any functions or methods.

The problem is that when another test includes the `sharing-service.php` (usually indirectly), the filters get added, but then reverted on `tearDown()`.
This leads to the situation when the files are included, classes and functions already defined, but the hooks required for them to work are empty.
So `require_once` doesn't happen obviously, we can't define those classes again.

This means that we need to add the filters again if the files were previously loaded.
I moved these two `add_filter()` calls into the `init()` method to make it easier to maintain.
The original `add_filter()` calls were run right in the `sharing-service.php`, so I decided not to change that behavior and added `Sharing_Service::init()` in their place. So when the file `sharing-service.php` is included, the filters are set right away.
That's not the best design choice, but changing that behavior requires modifying a lot of other components, and I feel like that's not a priority right now.
So I tried to change the code as little as possible to solve the problem.

This is a part of my `post-by-email` refactoring, by the way. My test indirectly loads the `sharing-service.php`, and breaks the test `WP_Test_Jetpack_Sync_Post::test_remove_sharedaddy_from_filtered_content()`.

Reply:
That makes sense, thanks for the explanation!